projects/extension/src/background/AppMediator.ts

```diff
@@ -45,10 +45,23 @@ export class AppMediator extends (EventEmitter as { new(): StateEmitter }) {
    * and the extension background.
    * @param manager - the extension's connection manager that keeps track of
    * all the apps and smoldots
+   * 
+   * @throws an error if the port name is not valid, in which case an error is
+   * sent on the port and the port gets disconnected
    */
   constructor(port: chrome.runtime.Port, manager: ConnectionManagerInterface) {
     super();
-    this.#appName = port.name.substr(0, port.name.indexOf('::'));
+
+    const splitIdx = port.name.indexOf('::');
+    if (splitIdx === -1) {
+      const error: MessageFromManager = { type: 'error', payload: `Invalid port name ${port.name} expected <app_name>::<chain_name>` };
+      port.postMessage(error);
+      port.disconnect();
+      throw new Error();
```

Comment:
Can we add a message to the error here and then `console.warn` in a `catch` block with the message below instead of using `finally` please?  Then we won't swallow unexpected error messages and we can debug problems locally more easily.

Reply:
I went for silencing the error because it's already sent back through the port to the user. The only advantage of also printing this to the console would be to debug situations where the code in the `ExtensionProvider` isn't working properly.

I think the code here should be redesigned to not directly send back the error to the port, in which case we'd indeed catch that error in order to send it back and not just discard it silently, and I'd do that on top of this port.

Reply:
The problem I want to avoid is that the `finally` below will swallow *all* errors.  If we warn in a `catch` block then we won't inadvertently swallow something unexpected.  Adding a message here helps disambiguate unexpected from programmer error by the app developer.